drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,436 @@
+	if (err < 0) {
+		return err;
+	}
+
+	/* Touch screen configuration */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_CFG_REG,
+				    stmpe811_tsc_config_bits(dev));
+	if (err < 0) {
+		return err;
+	}
+
+	/* Configure the touch FIFO threshold */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_TH_REG,
+				    STMPE811_FIFO_TH_SINGLE_POINT);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Clear the FIFO memory content */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG, STMPE811_FIFO_STA_CLEAR);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the range and accuracy of the pressure measurement (Z) */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_FRACT_XYZ_REG,
+				    STMPE811_TSC_FRACT_XYZ_CONF);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the driving capability (limit) of the device for TSC pins */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_I_DRIVE_REG,
+				    STMPE811_TSC_I_DRIVE_LIMIT);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Touch screen control configuration */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_CTRL_REG,
+				    stmpe811_tsc_control_bits(dev));
+	if (err < 0) {
+		return err;
+	}
+
+	/**
+	 * Clear all the status pending bits if any.
+	 * Writing '1' to this register clears the corresponding bits.
+	 * This is an 8-bit register, so writing 0xFF clears all.
+	 */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_INT_STA_REG, STMPE811_INT_ALL);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Put the FIFO back into operation mode */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG,
+				    STMPE811_FIFO_STA_OPERATIONAL);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Enable FIFO and touch interrupts */
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_INT_EN_REG,
+				    STMPE811_INT_BIT_TOUCH | STMPE811_INT_BIT_FIFO_THRESHOLD);
+	if (err < 0) {
+		LOG_ERR("Could not enable interrupt types (%d)", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int stmpe811_ts_get_data(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	uint8_t data_xy[3];
+	uint32_t uldata_xy;
+
+	int ret = i2c_burst_read_dt(&config->bus, STMPE811_TSC_DATA_NON_INC_REG, data_xy,
+				    sizeof(data_xy));
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Calculate positions values */
+	uldata_xy = (data_xy[0] << 16) | (data_xy[1] << 8) | data_xy[2];
+	data->touch_x = (uldata_xy >> 12U) & BIT_MASK(12);
+	data->touch_y = uldata_xy & BIT_MASK(12);
+
+	return 0;
+}
+
+static void stmpe811_report_touch(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	int x = config->screen_x_offset +
+		(data->touch_x * config->screen_width) / STMPE811_RAW_RES_XY;
+	int y = config->screen_y_offset +
+		(data->touch_y * config->screen_height) / STMPE811_RAW_RES_XY;
```

Comment:
That is super, thanks for doing it, just one request: can you change the binding default for screen height and width to 0 and then change this to only doing the calibration for `if (config->screen_width > 0 && config->screen_height > 0)`?

I'm thinking if we get another PR for a resistive touchscreen controller down the road, I'll ask for this function to be moved into a library file, so let's get away from a device specific default from now.

Reply:
This is now changed.